kotest-framework/kotest-framework-engine/src/commonMain/kotlin/io/kotest/core/spec/IsolationMode.kt

```diff
@@ -4,71 +4,13 @@ enum class IsolationMode {
 
    /**
     * A single instance of the [Spec] class is instantiated,
-    * and then used for every [io.kotest.core.test.TestCase].
-    *
-    * Test cases will be executed as soon as they are discovered.
+    * and then used for all [io.kotest.core.test.TestCase]s.
     */
    SingleInstance,
 
    /**
-    * A new instance of the [Spec] class is instantiated for every
-    * [io.kotest.core.test.TestCase] - both containers and leaf tests - and they are
-    * executed once the previous test has completed.
-    *
-    * For example, in the following test plan:
-    *
-    * ```
-    * "this test" {
-    *   println("a")
-    *   "nested test" {
-    *     println("b")
-    *   }
-    *   "nested test 2" {
-    *     println("c")
-    *   }
-    * }
-    * ```
-    *
-    * The output will be:
-    *
-    * ```text
-    * a
-    * a
-    * b
-    * a
-    * c
-    * ```
-    *
+    * A new instance of the [Spec] class is instantiated for every top level (root)
```

Comment:
I think we should keep the example code showing what is a root and what is not

For instance, I don't know if


Funspec
- Context1
    - -Test1
    - -Test2
- Context2
    - -Test3
    - -Test4


Will create 2 instances (Context1, Context2) or 3 instances (Funspec, Context1, Context2)

Or even a different thing altogether

Reply:
Comprehensive docs will follow, once it's in the final state.